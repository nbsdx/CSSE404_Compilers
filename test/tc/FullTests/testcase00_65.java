/*
 * Testcase designed to test unary expressions.
 * This should not pass, and the compiler should give errors to the effect of
 * "type mismatch".
 */

class Test8 {
	public static void main(String[] args) {
		boolean x = !3;
		int y = -false;
		int x = !true;
		boolean y = -50;
	}
}

// vim:filetype=java:textwidth=80
