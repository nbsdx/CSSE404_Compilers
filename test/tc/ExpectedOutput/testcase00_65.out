Operand type int is not compatible with expected type boolean of operator NOT
Operand type boolean is not compatible with expected type int of operator MINUS
The variable x is already declared in the current scope.
Cannot assign type boolean to variable x of type int
The variable y is already declared in the current scope.
Cannot assign type int to variable y of type boolean
